src/irmin/store.ml

```diff
@@ -903,6 +903,56 @@ module Make (P: S.PRIVATE) = struct
         | _ -> Lwt.return_none
       ) g
 
+  module Heap = Bheap.Make(struct
+      type t = commit * int
+      let compare c1 c2 =
+         Int64.compare
+          (Info.date (Commit.info (fst c1)))
+          (Info.date (Commit.info (fst c2)))
+    end)
+
+  let last_modified ?depth ?(number = 1) t key =
+    Head.get t >>= fun commit ->
```

Comment:
Can you add a debug message here?

Something like `Log.debug (fun l -> l "last_modified depth=%a number=%a" ...)`